src/custom/utils/metadata.ts

```diff
@@ -0,0 +1,38 @@
+interface IMetadata {
```

Comment:
My way of thinking is to always go with types whenever I need composition or if declared in a library as interfaces can be monkey patched. Here I chose an interface just because for me it was better explained as an inheritance and to keep consistency with the UML in the [doc](https://docs.google.com/document/d/1Zu-ygdI1QpsFrJp_Ux4Q5KQx6iQLRfVscjebaYB_RV0/edit#heading=h.majr50u2licw)